### ISSUE TITLE
aria-modal attribute set on wrong element?

### ISSUE THREAD

**Issue by kosmiq** [twbs/bootstrap#30516]
Hey all,

I've been looking into an [issue raised over at react-bootstrap](https://github.com/react-bootstrap/react-bootstrap/issues/5084). We have an issue stating that the attribute `aria-modal="true"` is added to the wrong element.

So I started looking into it on our end and realized we are doing it the same way as you are, adding it to the background element with the "modal" class.

Our issue raiser linked the following spec: https://amp.levelaccess.net/public/standards/view_best_practice.php?violation_id=1893 and then I looked into the w3 spec here: https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/dialog.html

Looking at the latter, and opening the example modal there the backdrop does not get either an `role="dialog"` attribute or `aria-modal="true"`.

So here's how Bootstrap and react-bootstrap does it:
![Screenshot 2020-04-06 at 10 21 36](https://user-images.githubusercontent.com/322759/78537909-7ced3c00-77f0-11ea-8719-f61f788725d7.png)

The backdrop styles element gets the aria attribute and `role="dialog"`
The example on w3 looks like this though:
![Screenshot 2020-04-06 at 10 21 54](https://user-images.githubusercontent.com/322759/78537965-92626600-77f0-11ea-95e1-1eb587cfc263.png)

Note that it is the dialog/modal itself that gets the attributes.

I found the following PR that added the attributes in 4.2.0: https://github.com/twbs/bootstrap/pull/27780

However, someone with more insight or experience with the aria and role specs for modals can perhaps shed some light on this? Are we adding the attributes to the wrong elements or has there been any discussions I've been unable to find regarding this?

**Comment by patrickhlauke** [comment]
is this just based on looking at the markup output, or is this causing any actual incorrect behavior when using assistive technologies?

conceptually, it makes no difference whether you say the whole outer container is the dialog (and set focus to it, and add all the relevant labelling/relationships to that outer container), or whether you do it on the inner container.

**Comment by patrickhlauke** [comment]
to be crystal, there is no functional difference between (simplifying it)

```
<div role="dialog" tabindex="-1" aria-modal="true" ...>
  <div>
    ...
  </div>
</div>
```

and

```
<div>
  <div role="dialog" tabindex="-1" aria-modal="true" ...>
     ...
  </div>
<div>
```

**Comment by kosmiq** [comment]
Thanks for replying @patrickhlauke. Well I haven't managed to find any issues with any a11y checks I've performed (hence the question mark in the title of the issue).

I agree that it should conceptually not be any differences. I wanted to raise the question after we got an issue on our end, and after looking at the w3c example and noticed the difference.

If you think that the current setup follows the spec and shouldn't cause any issues I'm all good :)
